driver/syscall_table.c

```diff
@@ -57,1937 +57,1267 @@ or GPL2.txt for full copies of the license.
+	[__NR_io_uring_register - SYSCALL_TABLE_ID0] =		{UF_USED, PPME_SYSCALL_IO_URING_REGISTER_E, PPME_SYSCALL_IO_URING_REGISTER_X, PPM_SC_IO_URING_REGISTER},
 #endif
 #ifdef __NR_copy_file_range
-	[__NR_copy_file_range - SYSCALL_TABLE_ID0] =            {UF_USED, PPME_SYSCALL_COPY_FILE_RANGE_E, PPME_SYSCALL_COPY_FILE_RANGE_X},
+	[__NR_copy_file_range - SYSCALL_TABLE_ID0] =            {UF_USED, PPME_SYSCALL_COPY_FILE_RANGE_E, PPME_SYSCALL_COPY_FILE_RANGE_X, PPM_SC_COPY_FILE_RANGE},
 #endif
 #ifdef __NR_open_by_handle_at
-	[__NR_open_by_handle_at - SYSCALL_TABLE_ID0] =                    {UF_USED, PPME_SYSCALL_OPEN_BY_HANDLE_AT_E, PPME_SYSCALL_OPEN_BY_HANDLE_AT_X},
+	[__NR_open_by_handle_at - SYSCALL_TABLE_ID0] =                    {UF_USED, PPME_SYSCALL_OPEN_BY_HANDLE_AT_E, PPME_SYSCALL_OPEN_BY_HANDLE_AT_X, PPM_SC_OPEN_BY_HANDLE_AT},
 #endif
 #ifdef __NR_mlock
-	[__NR_mlock - SYSCALL_TABLE_ID0] =			{UF_USED, PPME_SYSCALL_MLOCK_E, PPME_SYSCALL_MLOCK_X},
+	[__NR_mlock - SYSCALL_TABLE_ID0] =			{UF_USED, PPME_SYSCALL_MLOCK_E, PPME_SYSCALL_MLOCK_X, PPM_SC_MLOCK},
 #endif
 #ifdef __NR_munlock
-	[__NR_munlock - SYSCALL_TABLE_ID0] =			{UF_USED, PPME_SYSCALL_MUNLOCK_E, PPME_SYSCALL_MUNLOCK_X},
+	[__NR_munlock - SYSCALL_TABLE_ID0] =			{UF_USED, PPME_SYSCALL_MUNLOCK_E, PPME_SYSCALL_MUNLOCK_X, PPM_SC_MUNLOCK},
 #endif
 #ifdef __NR_mlockall
-	[__NR_mlockall - SYSCALL_TABLE_ID0] =			{UF_USED, PPME_SYSCALL_MLOCKALL_E, PPME_SYSCALL_MLOCKALL_X},
+	[__NR_mlockall - SYSCALL_TABLE_ID0] =			{UF_USED, PPME_SYSCALL_MLOCKALL_E, PPME_SYSCALL_MLOCKALL_X, PPM_SC_MLOCKALL},
 #endif
 #ifdef __NR_munlockall
-	[__NR_munlockall - SYSCALL_TABLE_ID0] =			{UF_USED, PPME_SYSCALL_MUNLOCKALL_E, PPME_SYSCALL_MUNLOCKALL_X},
+	[__NR_munlockall - SYSCALL_TABLE_ID0] =			{UF_USED, PPME_SYSCALL_MUNLOCKALL_E, PPME_SYSCALL_MUNLOCKALL_X, PPM_SC_MUNLOCKALL},
 #endif
 #ifdef __NR_mlock2
-	[__NR_mlock2 - SYSCALL_TABLE_ID0] = {UF_USED, PPME_SYSCALL_MLOCK2_E, PPME_SYSCALL_MLOCK2_X},
-#endif
-};
-
-/*
- * SYSCALL ROUTING TABLE
- */
-const enum ppm_syscall_code g_syscall_code_routing_table[SYSCALL_TABLE_SIZE] = {
-	[__NR_restart_syscall - SYSCALL_TABLE_ID0] = PPM_SC_RESTART_SYSCALL,
-	[__NR_exit - SYSCALL_TABLE_ID0] = PPM_SC_EXIT,
-	[__NR_read - SYSCALL_TABLE_ID0] = PPM_SC_READ,
-	[__NR_write - SYSCALL_TABLE_ID0] = PPM_SC_WRITE,
-	[__NR_execve - SYSCALL_TABLE_ID0] = PPM_SC_EXECVE,
-#ifdef __NR_open
-	[__NR_open - SYSCALL_TABLE_ID0] = PPM_SC_OPEN,
-#endif
-	[__NR_close - SYSCALL_TABLE_ID0] = PPM_SC_CLOSE,
-#ifdef __NR_open
-	[__NR_creat - SYSCALL_TABLE_ID0] = PPM_SC_CREAT,
-#endif
-#ifdef __NR_link
-	[__NR_link - SYSCALL_TABLE_ID0] = PPM_SC_LINK,
+	[__NR_mlock2 - SYSCALL_TABLE_ID0] = {UF_USED, PPME_SYSCALL_MLOCK2_E, PPME_SYSCALL_MLOCK2_X, PPM_SC_MLOCK2},
 #endif
-#ifdef __NR_unlink
-	[__NR_unlink - SYSCALL_TABLE_ID0] = PPM_SC_UNLINK,
-#endif
-	[__NR_chdir - SYSCALL_TABLE_ID0] = PPM_SC_CHDIR,
+	[__NR_restart_syscall - SYSCALL_TABLE_ID0] = { .ppm_sc = PPM_SC_RESTART_SYSCALL },
```

Comment:
From here, it starts the list of syscalls that were not preset in syscall_table, but were only present in routing table. We create them as empty with just the ppm_sc set.

Reply:
My rule of thumb to review this is: is Clion saying that every PPM_SC is used in syscall table? If yes, i did not forget anything. 
And that's the case :)